lib/improver/tests/weighted_blend/test_PercentileBlendingAggregator.py

```diff
@@ -0,0 +1,270 @@
+
+    def test_basic(self):
+        """Test that the __repr__ returns the expected string."""
+        result = str(PercentileBlendingAggregator())
+        msg = '<PercentileBlendingAggregator>'
+        self.assertEqual(result, msg)
+
+
+class Test_aggregate(IrisTest):
+    """Test the aggregate method"""
+    def test_blend_percentile_aggregate(self):
+        """Test blend_percentile_aggregate function works"""
+        weights = np.array([0.8, 0.2])
+        percentiles = np.array([0, 20, 40, 60, 80, 100])
+        result = PercentileBlendingAggregator.aggregate(
+            np.reshape(PERCENTILE_DATA, (6, 2, 2, 2)), 1,
+            percentiles,
+            weights, 0)
+        expected_result_array = np.reshape(BLENDED_PERCENTILE_DATA2,
+                                           (6, 2, 2))
+        self.assertArrayAlmostEqual(result, expected_result_array)
+
+    def test_blend_percentile_aggregate_reorder1(self):
+        """Test blend_percentile_aggregate works with out of order dims 1"""
+        weights = np.array([0.8, 0.2])
+        percentiles = np.array([0, 20, 40, 60, 80, 100])
+        perc_data = np.reshape(PERCENTILE_DATA, (6, 2, 2, 2))
+        perc_data = np.moveaxis(perc_data, [0, 1], [3, 1])
+        result = PercentileBlendingAggregator.aggregate(
+            perc_data, 1,
+            percentiles,
+            weights, 3)
+        expected_result_array = np.reshape(BLENDED_PERCENTILE_DATA2,
+                                           (6, 2, 2))
+        expected_result_array = np.moveaxis(expected_result_array, 0, 2)
+        self.assertArrayAlmostEqual(result, expected_result_array)
+
+    def test_blend_percentile_aggregate_reorder2(self):
+        """Test blend_percentile_aggregate works with out of order dims 2"""
+        weights = np.array([0.8, 0.2])
+        percentiles = np.array([0, 20, 40, 60, 80, 100])
+        perc_data = np.reshape(PERCENTILE_DATA, (6, 2, 2, 2))
+        perc_data = np.moveaxis(perc_data, [0, 1], [1, 2])
+        result = PercentileBlendingAggregator.aggregate(
+            perc_data, 2,
+            percentiles,
+            weights, 1)
+        expected_result_array = np.reshape(BLENDED_PERCENTILE_DATA2,
+                                           (6, 2, 2))
+        expected_result_array = np.moveaxis(expected_result_array, 0, 1)
+        self.assertArrayAlmostEqual(result, expected_result_array)
+
+    def test_2D_simple_case(self):
+        """ Test that for a simple case with only one point in the resulting
+            array the function behaves as expected"""
+        weights = np.array([0.8, 0.2])
+        percentiles = np.array([0, 50, 100])
+        perc_data = np.array([[1.0, 2.0], [5.0, 5.0], [10.0, 9.0]])
+        result = PercentileBlendingAggregator.aggregate(
+            perc_data, 1,
+            percentiles,
+            weights, 0)
+        expected_result = np.array([1.0, 5.0, 10.0])
+        self.assertArrayAlmostEqual(result, expected_result)
+
```

Comment:
It was mostly to do with checking the code worked with different numbers of dimensions. I removed a check that you had to make sure that the input cube had at least 3 dimensions, but so I wanted to check it did actually work with two input dimensions. But you're right it's not really checking any kind of sensible blending.